File: cross-application-feature-development-management/Names/FeatureName.cs
namespace cross_application_feature_development_management.Names
{
    public class FeatureName(CommandLineArgs commandLineArgs)
    {
        public string GetName()
        {
            const string featureNameKey = "--feature-name";
            var name = CommandLineArgs.GetByKey(featureNameKey);
            return name;
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Names/FeatureName.fs
namespace cafdemwimu.console.Names

open cafdemwimu.console
open cafdemwimu.console.Helpers

type FeatureName() =
    member _.GetName() =
        let featureNameKey = "--feature-name"
        let name = CommandLineArgs.GetByKey(featureNameKey)
        name

    member this.GetWrappedName(stringHelpers: StringHelpers) : string =
        let featureNameValue = this.GetName()
        let wrappedFeatureName = stringHelpers.WrapInQuotationMarks(featureNameValue)
        wrappedFeatureName
